detox/src/devices/runtime/drivers/android/AndroidDriver.js

```diff
@@ -148,7 +148,13 @@ class AndroidDriver extends DeviceDriverBase {
 
   async waitUntilReady() {
       try {
-        await Promise.race([super.waitUntilReady(), this.instrumentation.waitForCrash()]);
+        await Promise.race([
+          super.waitUntilReady(),
+          this.instrumentation.waitForCrash()
+        ]);
+      } catch (e) {
+        await this.client.waitUntilDisconnected();
```

Comment:
@noomorph does this feel safe for you? How about adding a custom timeout as a promise-race here? Or do you think that being jest-timeout bound is enough for this?

Reply:
Hmm I take it back. This can only happen if there was in fact a crash => app disconnected (assuming we don't have bugs inside `instrumentation.waitForCrash()`). Check me on that.